completion_aggregator/utils.py

```diff
@@ -28,3 +32,26 @@ def get_active_users(course_key):
     Return a list of users that have Aggregators in the course.
     """
     return get_user_model().objects.filter(aggregator__course_key=course_key).distinct()
+
+
+def make_datetime_timezone_unaware(date):
+    """
+    Return a timezone unaware(localize) version of the datetime instance.
+    """
+    # Reason: https://groups.google.com/forum/#!topic/django-developers/JA1uaAxnuGE
+    if DJANGO_MAJOR_VERSION >= 1 and DJANGO_MINOR_VERSION >= 10:
+        date = date.replace(tzinfo=None)
+    return date
```

Comment:
My reading of the linked post is that raw `SELECT` queries are returning timezone unaware values, but I don't understand why we need to force versions to be unaware before submitting them.  I suspect this will cause problems when retrieving aggregators normally, and trying to compare them with normal aware datetimes.  Did you run into actual issues when implementing this?

Reply:
@jcdyer 

Yeah, I ran into this error. Here are version specific logs

**Django version 2.0** (requires me to pass a naive datetime)
```
django.utils.deprecation.RemovedInDjango20Warning: The MySQL database adapter received an aware datetime (%s), probably from cursor.execute(). Update your code to pass a naive datetime in the database connection's time zone (UTC by default).
```

**Django version 1.8** (requires me to pass an aware datetime)
```
RuntimeWarning: MySQL received a naive datetime (2018-08-08 09:46:22.687201) while time zone support is active.
```
Also here is the log from one of the [travis job](https://travis-ci.org/open-craft/openedx-completion-aggregator/jobs/413493947) when this code was not present.
 

My reason for linking to that blog post, is I could not find a proper link which give details about this. Hence I posted that link because it was giving context that timezone unaware values are being returned.

Reply:
Hmm. Given:

1. https://docs.djangoproject.com/en/2.0/topics/i18n/timezones/#database
2. https://github.com/django/django/commit/e707e4c709c2e3f2dad69643eb838f87491891f8#diff-af003fcfed7cfbdeb396f8647ed0f92fR258
3. https://docs.djangoproject.com/en/2.1/ref/settings/#time-zone

I think we need to do `data.astimezone(timezone.utc).replace(tzinfo=None)` for Django 2+.

@jcdyer What do you think?

Reply:
Thanks for explaining @taranjeet.  That's helpful.  

@symbolist: The data *should* already be coming in with tzinfo=utc, but this would be safer, and should be a trivial amount of extra processing time. :+1: